src/file_pipe_log/log_file.rs

```diff
@@ -10,34 +10,45 @@ use log::warn;
 
 use crate::env::{FileSystem, Handle, WriteExt};
 use crate::metrics::*;
-use crate::pipe_log::{FileBlockHandle, LogFileContext, Version};
+use crate::pipe_log::{DataLayout, FileBlockHandle, LogFileContext};
+use crate::util::{round_down, round_up};
 use crate::{Error, Result};
 
-use super::format::LogFileFormat;
+use super::format::{LogFileFormat, LOG_FILE_HEADER_ALIGNMENT_SIZE};
 
 /// Maximum number of bytes to allocate ahead.
 const FILE_ALLOCATE_SIZE: usize = 2 * 1024 * 1024;
 
+/// Default alignment size.
+const FILE_ALIGNMENT_SIZE: usize = 32768; // 32kb as default
+
 /// Combination of `[Handle]` and `[Version]`, specifying a handler of a file.
 #[derive(Debug)]
 pub struct FileHandler<F: FileSystem> {
     pub handle: Arc<F::Handle>,
     pub context: LogFileContext,
 }
 
+#[inline]
+pub(crate) fn get_system_block_size() -> usize {
```

Comment:
The alignment size should be encoded in file. `Alignment(u64)`. We can later worry about the case when the system block size is different than the alignment size specified by log header.